site/gatsby-site/src/components/users/UsersField.js

```diff
@@ -0,0 +1,125 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useField } from 'formik';
+import { AsyncTypeahead, Token } from 'react-bootstrap-typeahead';
+import { useQuery } from '@apollo/client';
+import { FIND_USERS_FIELDS_ONLY } from '../../graphql/users';
+
+const filterBy = (option, text) => {
+  return (
+    option.first_name?.toLowerCase().includes(text.toLowerCase()) ||
+    option.last_name?.toLowerCase().includes(text.toLowerCase()) ||
+    option.id.toString().includes(text)
+  );
+};
+
+export default function UsersField({ id, name, placeHolder = '' }) {
+  const [{ value }, , { setTouched, setValue }] = useField({ name });
+
+  const { data } = useQuery(FIND_USERS_FIELDS_ONLY);
+
+  const [loading, setLoading] = useState(true);
+
+  const [options, setOptions] = useState([]);
+
+  const [selected, setSelected] = useState(
+    value
+      .slice()
+      .sort()
+      .map((id) => ({ id, first_name: '', last_name: '' }))
+  );
+
+  useEffect(() => {
+    if (data?.users) {
+      setSelected((selected) =>
+        selected.map(({ id }) => {
+          const { userId, first_name, last_name } = data.users.find((user) => user.userId == id);
+
+          return { id: userId, first_name, last_name };
+        })
+      );
+
+      setLoading(false);
+    }
+  }, [data]);
+
+  const handleSearch = useCallback(
+    async (text) => {
+      const options = data?.users
+        .filter(({ userId }) => !selected.some((s) => s.id == userId))
+        .map(({ userId: id, first_name, last_name }) => ({ id, first_name, last_name }))
+        .filter((option) => filterBy(option, text));
+
+      setOptions(options);
+    },
+    [data, selected]
+  );
+
+  return (
+    <>
+      <AsyncTypeahead
+        className="Typeahead incident-ids-field"
+        filterBy={() => true}
+        id={id}
+        inputProps={{ id: 'input-' + id, name }}
+        selected={selected}
+        options={options}
+        multiple
+        labelKey={(option) => `${option.first_name} ${option.last_name}`}
```

Comment:
Do we have a way to avoid these code errors?
<img width="729" alt="image" src="https://github.com/responsible-ai-collaborative/aiid/assets/6564809/9a84ce30-c2d5-4439-8664-e67eb9a9d526">

Reply:
We would have to use a jsdoc comment or switch to typescript. Not worth it at the moment.